### PR TITLE
SoF - Thursagan as "effect crafter"

### PR DESCRIPTION
**Pull request by doofus-01**
_Initial commit of Thursagan rune-shop addition_
This is to deal with the issues in https://forums.wesnoth.org/viewtopic.php?f=38&p=650382
Rather than a new unit type, resources can be spent to temporarily (scenario) buff a standing unit.   
 - Costs are supposed to make it competitive with just recruiting another unit.
 - UI is probably the first issue to tune, I tried to make it more available than just an obscure [set_menu_item], without being too annoying.
 - Balance will be an issue for a while, probably, but it should be easy to adjust code-wise.

### CODE REVIEW

**Comment by doofus-01** [comment]
Currently, all [item] halos get removed, it seems line "[remove_item]name=" key does no filtering, just removes everything.  Does anyone know of a working example ([remove_item] using name key, not x,y)?  If x,y is required, the "name" key is pretty useless - just use the image/halo filename, finding x,y is the hard part.

**Comment by doofus-01** [comment]
I believe this is fully working now, though there is room for improved graphics, balance adjustments, and maybe more rune options.  I'll merge it next weekend, if no one has objections.

File: data/campaigns/Sceptre_of_Fire/utils/rune-equip.cfg [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+    [filter_condition]
+        [variable]
+            name=chest_{X}_{Y}
+            equals=yes
+        [/variable]
+    [/filter_condition]
+        {CLEAR_VARIABLE chest_{X}_{Y}}
+        [message]
+            speaker=unit
+            message= _ "This is part of our stash of field runes, we can use them to get a small temporary boost.  We don't have very many though, and need to contribute funds for replacements.  should I take this one?"
```
Comment (ln-zookeeper) [inline data/campaigns/Sceptre_of_Fire/utils/rune-equip.cfg]: Missing capitalization for "should". Also, double spaces here and in all other messages.

**Comment by Pentarctagon** [comment]
The WML schema validation is failing: https://travis-ci.com/wesnoth/wesnoth/jobs/272590418#L424

**Comment by CelticMinstrel** [comment]
The `halo=` key does not do anything in `[remove_item]` so I guess it can just be removed.

**Comment by doofus-01** [comment]
Thanks, I got distracted by #4689, forgot I still had halo keys in there.

**Comment by gfgtdf** [comment]
using locations instead will also remove all other images from that location, please use [remove_item] image="rune_shop_overlay" [/remove_item] (like you did before but with image= instead of name= in [remove_item]) instead or additionally.

**Comment by gfgtdf** [comment]
actuyll, consdering that this is 1.15, you can also change the [remove_item] syntax in the engine if you really don't like  that it calles name= in [item] but image= in [remove_item]

**Comment by doofus-01** [comment]
OK.  Changing the engine should probably be in a different PR though.

**Pull request merged**